packages/plugins/i18n/admin/src/components/CMHeaderActions.tsx

```diff
@@ -226,6 +236,302 @@ const DeleteLocaleAction: DocumentActionComponent = ({
+  const {
+    document,
+    meta: documentMeta,
+    schema,
+    validate,
+  } = useDocument({
+    model,
+    collectionType,
+    documentId,
+    params: {
+      locale: baseLocale,
+    },
+  });
+
+  const { data: localesMetadata = [] } = useGetLocalesQuery();
+
+  const headers = [
+    {
+      label: formatMessage({
+        id: 'global.name',
+        defaultMessage: 'Name',
+      }),
+      name: 'name',
+    },
+    {
+      label: formatMessage({
+        id: getTranslation('CMEditViewBulkLocale.status'),
+        defaultMessage: 'Status',
+      }),
+      name: 'status',
+    },
+    {
+      label: formatMessage({
+        id: getTranslation('CMEditViewBulkLocale.publication-status'),
+        defaultMessage: 'Publication Status',
+      }),
+      name: 'publication-status',
+    },
+  ];
+
+  const [rows, validationErrors] = React.useMemo(() => {
+    if (!document || !documentMeta?.availableLocales) {
+      return [[], {}];
+    }
+
+    const rowsFromMeta: LocaleStatus[] = documentMeta?.availableLocales.map((doc) => {
+      const { locale, status } = doc;
+
+      return { locale, status };
+    });
+    rowsFromMeta.unshift({
+      locale: document?.locale ?? '',
+      status: document?.status ?? 'draft',
+    });
+
+    const allDocuments = [document, ...(documentMeta?.availableLocales ?? [])];
+
+    const errors: Record<
+      Modules.Documents.Params.Locale.StringNotation,
+      Record<string, MessageDescriptor>
+    > = {};
+
+    allDocuments.map((document) => {
+      if (!document) {
+        return document;
+      }
+
+      const validation = validate(document as Modules.Documents.AnyDocument);
+      if (validation !== null) {
+        errors[document.locale] = validation;
+      }
+
+      return document;
+    });
+
+    return [rowsFromMeta, errors];
+  }, [document, documentMeta?.availableLocales, validate]);
+
+  const localesToPublish = selectedRows
+    .filter(
+      (selectedRow) =>
+        selectedRow.status !== 'published' &&
+        !Object.keys(validationErrors || {}).includes(selectedRow.locale)
+    )
+    .map((selectedRow) => selectedRow.locale);
+
+  const {
+    data: draftRelationsCount = 0,
+    isLoading: isDraftRelationsLoading,
+    error: isDraftRelationsError,
+  } = useGetManyDraftRelationCountQuery(
+    {
+      model,
+      documentIds: [documentId!],
+      locale: localesToPublish,
+    },
+    {
+      skip: !documentId || localesToPublish.length === 0,
+    }
+  );
+
+  React.useEffect(() => {
+    if (isDraftRelationsError) {
+      toggleNotification({
+        type: 'danger',
+        message: formatAPIError(isDraftRelationsError),
+      });
+    }
+  }, [isDraftRelationsError, toggleNotification, formatAPIError]);
+
+  if (!schema?.options?.draftAndPublish ?? false) {
+    return null;
+  }
+
+  if (!hasI18n) {
+    return null;
+  }
+
+  if (!documentId) {
+    return null;
+  }
+
+  if (!canPublish) {
+    return null;
+  }
```

Comment:
I think if you can't do the action, we disable it? but if the action shouldn't be visible e.g. no D&P / no i18n then we return null 🤔 iirc....